app/src/main/java/com/termux/app/TermuxInstaller.java

```diff
@@ -209,18 +209,22 @@ private static String determineTermuxArchName() {
             Arrays.toString(Build.SUPPORTED_ABIS));
     }
 
-    /** Delete a folder and all its content or throw. */
+    /** Delete a folder and all its content or throw. Don't follow symlinks. */
     static void deleteFolder(File fileOrDirectory) throws IOException {
-        File[] children = fileOrDirectory.listFiles();
-        if (children != null) {
-            for (File child : children) {
-                if (child.getCanonicalFile().equals(child.getAbsoluteFile())) {
-                    deleteFolder(child);
-                } else {
-                    child.delete();
+        if (fileOrDirectory.getCanonicalFile().equals(fileOrDirectory.getAbsoluteFile())) {
+            File[] children = fileOrDirectory.listFiles();
+
+            if (children != null) {
+                for (File child : children) {
+                    if (child.getCanonicalFile().equals(child.getAbsoluteFile())) {
```

Comment:
Here you only need to do `isDirectory` as the Canonical vs Absolute test will be perform in the `deleteFolder` call that follows.